src/viewport/src/Client/Viewport.lua

```diff
@@ -253,7 +265,16 @@ function Viewport:Render(props)
 			end);
 		[Blend.OnChange "AbsoluteSize"] = self._absoluteSize;
 		[Blend.Attached(function(viewport)
-			return ViewportControls.new(viewport, self)
+			local controlsMaid = Maid.new()
+
+			-- create viewport controls and obey enabled state
+			local viewportControls = ViewportControls.new(viewport, self)
```

Comment:
might technically be nicer to construct only when necessary, but this code is fine tbh.